api/src/main/java/org/pmiops/workbench/db/dao/DataSetServiceImpl.java

```diff
@@ -660,14 +660,25 @@ private static String generateNotebookUserCode(
             .append(domainAsString)
             .append("\"")
             .toString();
+    String sqlLimitSection =
+        new StringBuilder(
+                "# The ‘max_number_of_rows’ parameter limits the number of rows in the query so that the result set can fit in memory.\n")
+            .append(
+                "# If you increase the limit and run into responsiveness issues, please request a VM size upgrade.\n")
+            .append("max_number_of_rows = '1000000'")
```

Comment:
I went with a string instead of an integer since it simplifies the string concatenation code. It's especially a problem in R because the default toString() on integers uses scientific notation, requiring the use of a format function.

Reply:
BTW, not saying this would be applicable in a bug time frame, but I have some thoughts around making the SQL generation and Python/R generation more robust and flexible. I'd like to get your thoughts.

See https://precisionmedicineinitiative.atlassian.net/browse/RW-3340 and https://precisionmedicineinitiative.atlassian.net/browse/RW-3466